### ISSUE TITLE
[Question] Is it possible to generate with transparent background using this plugin?

### ISSUE THREAD

**Issue by Drakmour** [isekaidev/stable.art#45]
I know that Stable Diffusion doesn't know yet what alpha chanel is and can't make transparency. But considering that it's a photoshop plugin, maybe it can now? So is it possible to generate with transparent background using this plugin? Or maybe will be possible?

**Comment by GBaige** [comment]
stable.art is just a front-end for the webui, so it can do what the webui allows it to. The webui cannot do what you're asking. The next best thing is specifying in the prompt that you want a solid white background (maybe even a solid green background that you can chroma-key out later).

**Comment by Drakmour** [comment]
I just hoped that Photoshop could use some of it's own tricks in generations. :-) I hope SD will soon add transparency, as I heard they work this idea out.

**Comment by GBaige** [comment]
Perhaps the plugin could query a script to automate the process of removing a background but results would vary depending on the colors/content of the generated image. Remember, Photoshop itself isn't doing anything more than displaying the generation's results since everything happens in the webui/backend.

**Comment by Drakmour** [comment]
> Perhaps the plugin could query a script to automate the process of removing a background but results would vary depending on the colors/content of the generated image. Remember, Photoshop itself isn't doing anything more than displaying the generation's results since everything happens in the webui/backend.

Nah, it would be low quality, Same as just remove that background with magic eraser. Well, we'll wait until SD will learn the alpha chanel.

**Comment by GBaige** [comment]
We'll wait a very long time then.

The diffusion model works by adding then removing noise from an image, attempting to create objects out of the resulting information for every specified step. For txt2img, the starting image is random color noise. There's no way for the model to determine what will become transparent during the first stage, so it makes more sense to manually remove the background after the generation is finished.

**Comment by Drakmour** [comment]
> maybe even a solid green background that you can chroma-key out later

Btw as long as you mentioned it. I know that it is possible to remove the chroma-key, but is it more precise than any other color? I saw some videos where people clear the image from the green but they did the same as they would do with any other color. Just determine a color that will be deleted, what so special about the green and blue chromakey? As I know there is no "magic" tool that will remove the green ideally clean. You will clean the edges anyway whether you removed green or any other color.

**Comment by GBaige** [comment]
> Just determine a color that will be deleted.

Chroma-keying is a little more involved than simply selecting a specific colour and then deleting it. It's not a magic wand-like tool; it looks for color information throughout the whole image based on a specified color _range_.

> what so special about the green and blue chromakey?

Nothing, except green screens and blue screens (especially green screens) are popular because the subject is less likely to contain the color information from the background that will be keyed out. In digital applications where alpha information is not supported, magenta (#ff00ff) as a background is very popular.

> As I know there is no "magic" tool that will remove the green ideally clean. You will clean the edges anyway whether you removed green or any other color.

Ideally you want the subject to be very sharp against the background to be keyed out. In this case, the stable diffusion model is capable of producing very sharp images of subjects against specified solid color backgrounds, such that in some cases no keying would be required. Even still, in a lot of cases tweaking is still necessary.

All in all, the keying quality will depend on the quality of the image itself. You can't polish a turd.

**Comment by Drakmour** [comment]
So I guess there is not much difference of what color will be the background of the generated image. Green or yellow or white it will depend on how sharp edges do your needed image has. :-D I guess that's the only way, same old magic eraser and simple eraser "finetune". :-)

**Comment by GBaige** [comment]
I don't think you're understanding me. Chroma keying is not comparable to using a magic eraser or erasing the background by hand. It looks for color information. If for instance you generated an image of a character with a yellow background and you attempt to chroma key the yellow out of the image, the character itself will be affected because the colors of the character likely have some yellow mixed into them (the skin, for instance). You'll end up with a character with translucent skin.

Anyway, this conversation is going off-topic from the original post. You already have your answer.
